src/app.js

```diff
@@ -40,7 +41,9 @@ const getMenuItemNames = (limit = null) => {
    * Can you implement a limit here? I bet you can...
    */
 
-  return allNames;
+  const namesLimited = allNames.slice(0,limit);
+
+  return limit ? namesLimited : allNames;
```

Comment:
Here, I would advise only performing the `.slice()` operation if there is a limit. If there is not a need for it, then it is more optimal to return `allNames`. For example, you could do:
```
return !limit ? allNames : allNames.slice(0,limit);
```

Reply:
Oh yes, you are totally right! 🙏  Thanks!